craft_application/project.py

```diff
@@ -23,37 +54,27 @@ class Config:  # pylint: disable=too-few-public-methods
         alias_generator = lambda s: s.replace("_", "-")  # noqa: E731
 
 
-# A workaround for mypy false positives
-# see https://github.com/samuelcolvin/pydantic/issues/975#issuecomment-551147305
-# fmt: off
-if TYPE_CHECKING:
-    UniqueStrList = List[str]
-else:
-    UniqueStrList = conlist(str, unique_items=True)
-# fmt: on
-
-
 class Project(ProjectModel):
     """Craft Application project definition."""
 
-    name: constr(max_length=40)  # type: ignore
-    title: Optional[constr(max_length=40)]  # type: ignore
-    base: Optional[str]
-    build_base: Optional[str]
-    version: constr(max_length=32, strict=True)  # type: ignore
+    name: ProjectName
+    title: Optional[ProjectTitle]
+    base: Optional[ProjectBaseStr]
+    build_base: Optional[ProjectBaseStr]
+    version: VersionStr
     contact: Optional[Union[str, UniqueStrList]]
     donation: Optional[Union[str, UniqueStrList]]
     issues: Optional[Union[str, UniqueStrList]]
-    source_code: Optional[str]
-    website: Optional[str]
-    summary: Optional[constr(max_length=78)]  # type: ignore
+    source_code: Optional[AnyUrl]
+    website: Optional[AnyHttpUrl]
+    summary: Optional[SummaryStr]
     description: Optional[str]
     license: Optional[str]
     parts: Dict[str, Any]  # parts are handled by craft-parts
 
     @pydantic.validator("name")
     @classmethod
-    def _validate_name(cls, name):
+    def _validate_name(cls, name: str) -> str:
```

Comment:
now that `name` is a `ProjectName`, is this validator still used?

Reply:
Not entirely sure! I'll take a look and check, because it does seem like it should.